lib/src/atoms/command/exec.rs

```diff
@@ -43,20 +43,35 @@ impl Exec {
 
             // Requested priviledged, but is not root
             (true, _) => (
-                String::from("sudo"),
+                self.sudo_path().unwrap(),
                 [vec![self.command.clone()], self.arguments.clone()].concat(),
             ),
         }
     }
 
+    fn sudo_path(&self) -> anyhow::Result<String> {
```

Comment:
One thing I don't like is that `sudo_path()` doesn't cache it's results, so it's not as efficient as it could be. I'm still learning Rust, so I'm not sure how best to do this.